go/ai/retriever.go

```diff
@@ -23,12 +23,28 @@ import (
 )
 
 type (
-	// An Indexer is used to index documents in a store.
-	Indexer core.Action[*IndexerRequest, struct{}, struct{}]
-	// A Retriever is used to retrieve indexed documents.
-	Retriever core.Action[*RetrieverRequest, *RetrieverResponse, struct{}]
+	// An IndexerAction is used to index documents in a store.
+	IndexerAction core.Action[*IndexerRequest, struct{}, struct{}]
+	// A RetrieverAction is used to retrieve indexed documents.
+	RetrieverAction core.Action[*RetrieverRequest, *RetrieverResponse, struct{}]
 )
 
+// Retriever represents a document retriever.
+type Retriever interface {
+	// Name returns the name of the retriever.
+	Name() string
```

Comment:
If this has Name, then  Model should too.

Reply:
Added to model.